src/mode/modeHandler.ts

```diff
@@ -1446,7 +1448,7 @@ export class ModeHandler implements vscode.Disposable {
             await this.handleMultipleKeyEvents(keyStrokes);
           }
 
-          vimState.isReplayingMacro = false;
+          vimState.isReplayingMacro --;
```

Comment:
Wait, I'm not sure I want to merge this yet. Although this does fix the bug mentioned, I think the other PR I submitted for this issue(https://github.com/VSCodeVim/Vim/pull/1635) may be a better long term solution. I have a lot of work coming up this week so I'll take another look in a week or two.